### PR TITLE
Add a forEach to test each schema when there is an allOf

### PR DESCRIPTION
**Pull request by keldorus**
Respond to issue #64 

Feel free to comment and argue, I don't open PR often.

### CODE REVIEW

File: test/allof.test.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+const test = require('tap').test
+const build = require('..')
+
+test('object with multiple types field', (t) => {
+  t.plan(2)
+
+  const schema = {
+    title: 'object with multiple types field',
+    type: 'object',
+    allOf: [
```
Comment (mcollina) [inline test/allof.test.js]: can you add a test for an empty `allOf` and one with one element.

**Comment by keldorus** [comment]
Hum maybe I can add a last test for the required part, no?

**Comment by mcollina** [comment]
Yes

**Comment by keldorus** [comment]
ok done.
